reference_entities/add_custom_property_to_your_custom_attribute_type.rst

```diff
@@ -0,0 +1,867 @@
+        return this.unit;
+      }
+    }
+
+    /**
+     * This type is an aggregate of all the custom properties. Here we only have one so it could seems useless but
+     * here is an example with multiple properties:
+     *
+     *     export type TextAdditionalProperty = MaxLength | IsTextarea | IsRichTextEditor | ValidationRule | RegularExpression;
+     *
+     * In the example above, a additional property of a text attribute could be a Max length, is textarea, is rich text editor, ...
+     */
+    export type SimpleMetricAdditionalProperty = MetricUnit;
+
+    /**
+     * Same for the non normalized form
+     */
+    export type NormalizedSimpleMetricAdditionalProperty = NormalizedMetricUnit;
+
+    /**
+     * This interface will represent your normalized attribute (usually coming from the backend but also used in the reducer)
+     */
+    export interface NormalizedSimpleMetricAttribute extends NormalizedAttribute {
+      type: 'simple_metric';
+      unit: NormalizedMetricUnit; // add the custom property
+    }
+
+    /**
+     * Here we define the interface for our concrete class (our model) extending the base attribute interface
+     */
+    export interface SimpleMetricAttribute extends Attribute {
+      unit: MetricUnit; // add the custom property
+      normalize(): NormalizedSimpleMetricAttribute;
+    }
+
+    /**
+     * Here we are starting to implement our custom attribute class.
+     * Note that most of the code is due to the custom property (defaultValue). If you don't need to add a
+     * custom property to your attribute, the code can be stripped to it's minimal
+     */
+    export class ConcreteSimpleMetricAttribute extends ConcreteAttribute implements SimpleMetricAttribute {
+      /**
+       * Here, our constructor is private to be sure that our model will be created through a named constructor
+       */
+      private constructor(
+        identifier: Identifier,
+        referenceEntityIdentifier: ReferenceEntityIdentifier,
+        code: AttributeCode,
+        labelCollection: LabelCollection,
+        valuePerLocale: boolean,
+        valuePerChannel: boolean,
+        order: number,
+        is_required: boolean,
+        readonly unit: MetricUnit // change our constructor
+      ) {
+        super(
+          identifier,
+          referenceEntityIdentifier,
+          code,
+          labelCollection,
+          'simple_metric',
+          valuePerLocale,
+          valuePerChannel,
+          order,
+          is_required
+        );
+
+        /**
+         * Always ensure that your object is well formed from it's constructor to avoid crash of the application
+         */
+        if (!(unit instanceof MetricUnit)) {
+          throw new Error('Attribute expect a MetricUnit as unit');
+        }
+
+        /**
+         * This will ensure that your model is not modified after it's creation (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze)
```

Comment:
```suggestion
         * This will ensure that your model is not modified after its creation (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze)
```